actors/power/tests/power_actor_tests.rs

```diff
@@ -1095,6 +1095,93 @@ mod cron_batch_proof_verifies_tests {
         rt.verify();
         h.check_state();
     }
+
+    #[test]
+    fn skips_verify_if_miner_has_no_claim() {
+        let (mut h, mut rt) = setup();
+        h.create_miner_basic(&mut rt, OWNER, OWNER, MINER_1).unwrap();
+
+        let info = create_basic_seal_info(1);
+
+        h.submit_porep_for_bulk_verify(&mut rt, MINER_1, info).unwrap();
+
+        h.delete_claim(&mut rt, &MINER_1);
+
+        let infos = vec![];
+
+        let confirmed_sectors = vec![];
+
+        h.on_epoch_tick_end(&mut rt, 0, &BigInt::zero(), confirmed_sectors, infos);
+
+        h.check_state();
+    }
+
+    #[test]
+    fn success_with_multiple_miners_and_multiple_confirmed_sectors_and_assert_expected_power() {
+        let miner1 = Address::new_id(101);
+        let miner2 = Address::new_id(102);
+        let miner3 = Address::new_id(103);
+        let miner4 = Address::new_id(104);
+
+        let info1 = create_basic_seal_info(1);
+        let info2 = create_basic_seal_info(2);
+        let info3 = create_basic_seal_info(3);
+        let info4 = create_basic_seal_info(101);
+        let info5 = create_basic_seal_info(200);
+        let info6 = create_basic_seal_info(201);
+        let info7 = create_basic_seal_info(300);
+        let info8 = create_basic_seal_info(301);
+
+        let (mut h, mut rt) = setup();
+
+        h.create_miner_basic(&mut rt, OWNER, OWNER, miner1).unwrap();
+        h.create_miner_basic(&mut rt, OWNER, OWNER, miner2).unwrap();
+        h.create_miner_basic(&mut rt, OWNER, OWNER, miner3).unwrap();
+        h.create_miner_basic(&mut rt, OWNER, OWNER, miner4).unwrap();
+
+        h.submit_porep_for_bulk_verify(&mut rt, miner1, info1.clone()).unwrap();
+        h.submit_porep_for_bulk_verify(&mut rt, miner1, info2.clone()).unwrap();
+
+        h.submit_porep_for_bulk_verify(&mut rt, miner2, info3.clone()).unwrap();
+        h.submit_porep_for_bulk_verify(&mut rt, miner2, info4.clone()).unwrap();
+
+        h.submit_porep_for_bulk_verify(&mut rt, miner3, info5.clone()).unwrap();
+        h.submit_porep_for_bulk_verify(&mut rt, miner3, info6.clone()).unwrap();
+
+        h.submit_porep_for_bulk_verify(&mut rt, miner4, info7.clone()).unwrap();
+        h.submit_porep_for_bulk_verify(&mut rt, miner4, info8.clone()).unwrap();
+
+        let confirmed_sectors = vec![
```

Comment:
please copy over the comment "Because read order of keys..."

Reply:
👍